app/views/tasks/show.html.slim

```diff
@@ -4,40 +4,40 @@
       .std-heading data-bs-toggle='tooltip' data-bs-placement='bottom' title=@task.title
         i.i.fa-regular.fa-file-lines
         =< truncate(@task.title, length: 100)
-      span.task-star-rating
-        - [*1..5].each do |i|
-          - if @task.average_rating >= i
-            span.fa-solid.fa-star.overall-rating data-rating=i
-          - elsif (@task.average_rating + 0.5) >= i
-            span.fa-regular.fa-star-half-stroke.overall-rating data-rating=i
-          - else
-            span.fa-regular.fa-star.overall-rating data-rating=i
-        span
-          .rating-dropdown
-            - if current_user && policy(Rating).new?
-              .btn#rate
-                => t('.button.rate')
-                span.fa-solid.fa-caret-down
-
-              .popup-rating.shadow-sm.bg-light
-                .rating data-task-id=@task.id
-                  - if @user_rating
-                    - [*1..5].each do |i|
-                      - if @user_rating >= i
-                        span.fa-solid.fa-star data-rating=i
-                      - else
-                        span.fa-regular.fa-star data-rating=i
-                  - else
-                    - [*1..5].each do |i|
-                      span.fa-regular.fa-star data-rating=i
 
-            - else
-              div[data-bs-toggle='tooltip' title=t('.guest.disabled.tooltip') data-bs-delay=150]
-                .btn#rate
-                  => t('.button.rate')
-                  span.fa-solid.fa-caret-down
+      .d-flex.align-items-center
+        span.task-star-rating.rating-subcategories-trigger.averaged-task-ratings
+          = render('task_star_rating', rating: @task.overall_rating, category: :overall_rating)
+          .rating-subcategories-overlay
+            = render('rating_subcategories', task: @task)
+        span.mx-2
+          - if current_user && policy(Rating).new_for_task?(@task)
+            .btn.btn-light.btn-sm#ratingModalOpenButton data-bs-toggle="modal" data-bs-target="#ratingModal"
+              = t('.button.rate')
+            .modal.fade#ratingModal aria-hidden="true" data-task-id=@task.id
```

Comment:
Maybe, you could consider to extract the modal to a dedicated partial, allowing us to reuse it later? For example, similar to the [structure found in CodeOcean](https://github.com/openHPI/codeocean/blob/master/app/views/shared/_modal.html.slim) (with the necessary adaptions you made already for CodeHarbor).

Reply:
Thanks for extracting the modal as requested, but I meant something different (even if that's not too clear when reading my own comment) 🙈.

Let's keep the extracted partial you've created, it reduces the overall length of the `tasks/show.html.slim` file.

What I meant was to have a *generic* partial for any modal, that could be instantiated multiple times. For example, imagine there is a file called `app/views/shared/_modal.html.slim` which accepts multiple parameters. Then, you could simply do something like:

```slim
= render('shared/modal', id: 'ratingModal', title: Rating.model_name.human, template: 'tasks/_rating_modal')
```

The partial `app/views/shared/_modal.html.slim` would contain everything that is generally needed for all modals, such as setting the `zindex`, Bootstrap classes, etc. And in `app/views/tasks/_rating_modal.html.slim`, you would simply add the *content* of the modal only and potentially content for the footer (if any). This way, maintaining changes to Bootstrap and the views is easier to implement. The example I linked from CodeOcean is this generic file, with [this one](https://github.com/openHPI/codeocean/blob/092e8426671324db7489378c1f0d17b344ee58b5/app/views/programming_groups/_info_pair_programming.html.slim) being an example for a specific template (modal content).

Do you think my proposal makes sense?

Reply:
Yes, that makes sense but I implemented it a bit different to CodeOcean to avoid needing a new partial for every different modal body. What do you think?

Reply:
Yes, that's fine for me. I've made a few code suggestions there. Also, it currently feels a bit weird to have an empty button in the footer -- let's simply move the "Save" button there.

<img width="1840" alt="Bildschirmfoto 2024-08-17 um 18 38 14" src="https://github.com/user-attachments/assets/45023cb4-e9a3-47fa-8a08-4b321970bc98">

Reply:
Sorry, I somehow missed that...

Reply:
Don't worry, thanks for taking care. I can confirm the button is gone 😉